### PR TITLE
Allow auth to accept API keys

### PR DESCRIPTION
**Pull request by saiatmakuri**
first step in API key migration - chain functions tried

### CODE REVIEW

**Review by seanshi-scale — COMMENTED** [review]
I'm not super confident we should be exposing notions e.g. user id + api key for the public repo, e.g. this functionality may belong better somewhere else (e.g. `model-engine-internal`)

**Comment by yixu34** [comment]
> I'm not super confident we should be exposing notions e.g. user id + api key for the public repo, e.g. this functionality may belong better somewhere else (e.g. `model-engine-internal`)

Hmm I suppose that may be true. If we're hosting internally, we don't need to mention API keys here because the rest of our stack handles that; if we're deploying in customer VPCs, we rely on them to manage API keys and auth.

The main difference is that the EGP APIs do need to do this, because they're more of a fully fledged platform product that needs to manage its own API keys. cc @eric-scale

**Comment by saiatmakuri** [comment]
> I'm not super confident we should be exposing notions e.g. user id + api key for the public repo, e.g. this functionality may belong better somewhere else (e.g. model-engine-internal)
I guess that I'm just acknowledging the underlying `auth_repo` functions here more loudly. That is, we currently use `auth_repo.get_auth_from_user_id_async` and now I'm referencing `auth_repo.get_auth_from_api_key_async` in a separate routine. The final steady state will be to only use the latter method with api keys. I believe we can introduce a new notion of `UserRepo` that can perform authentication via API keys. However, I think we defer this abstraction to a later time.

**Review by yixu34 — APPROVED** [review]
【﻿ｗｅ　ｌｏｖｅ　ｄｅｐｅｎｄｅｎｃｙ　ｉｎｊｅｃｔｉｏｎ】

File: model-engine/model_engine_server/infra/gateways/datadog_monitoring_metrics_gateway.py [hunk 1]
```diff
@@ -1,38 +0,0 @@
-from datadog import statsd
-from model_engine_server.core.config import infra_config
-from model_engine_server.domain.gateways.monitoring_metrics_gateway import MonitoringMetricsGateway
-
-
-class DatadogMonitoringMetricsGateway(MonitoringMetricsGateway):
```
Comment (seanshi-scale) [inline model-engine/model_engine_server/infra/gateways/datadog_monitoring_metrics_gateway.py:6]: did this file get moved to internal?

Comment (yixu34) [reply model-engine/model_engine_server/infra/gateways/datadog_monitoring_metrics_gateway.py:6]: I suppose that dependencies on public vendors (e.g. Datadog) can be put in our public repo. It's the Scale-specific bits that need to be internal.

Comment (saiatmakuri) [reply model-engine/model_engine_server/infra/gateways/datadog_monitoring_metrics_gateway.py:6]: yup, now a plugin

**Pull request merged**